src/hoverifier.ts

```diff
@@ -801,12 +823,134 @@ export function createHoverifier<C extends object, D, A>({
+                    position,
+                    documentHighlightsOrError,
+                    part: position.part,
+                })),
+                // Do not emit anything after the code view this action came from got unhoverified
+                takeUntil(allUnhoverifies.pipe(filter(unhoverifiedCodeViewId => unhoverifiedCodeViewId === codeViewId)))
+            )
+        }),
+        share()
+    )
+
+    // Highlight the ranges returned by the document highlight provider
+    subscription.add(
+        documentHighlightObservables
+            .pipe(
+                switchMap(highlightObservable => highlightObservable),
+                switchMap(({ documentHighlightsOrError, position, adjustPosition, codeView, part, ...rest }) => {
+                    const highlights =
+                        documentHighlightsOrError &&
+                        documentHighlightsOrError !== LOADING &&
+                        !isErrorLike(documentHighlightsOrError)
+                            ? documentHighlightsOrError
+                            : []
+
+                    if (highlights.length === 0 || !position) {
+                        return of({ adjustPosition, codeView, part, ...rest, positions: of<Position[]>([]) })
+                    }
+
+                    const positions: Observable<Position>[] = []
+                    for (const { range } of highlights) {
+                        let pos = { ...position, ...range.start }
+
+                        // The requested position is is 0-indexed; the code here is currently 1-indexed
+                        const { line, character } = pos
+                        pos = { ...pos, line: line + 1, character: character + 1 }
+
+                        positions.push(
+                            adjustPosition
+                                ? from(
+                                      adjustPosition({
+                                          codeView,
+                                          direction: AdjustmentDirection.ActualToCodeView,
+                                          position: {
+                                              ...pos,
+                                              part,
+                                          },
+                                      })
+                                  )
+                                : of(pos)
+                        )
+                    }
+
+                    return of({
+                        adjustPosition,
+                        codeView,
+                        part,
+                        ...rest,
+                        positions: combineLatest(positions),
+                    })
+                }),
+                mergeMap(({ positions, codeView, dom, part }) =>
+                    positions.pipe(
+                        map(highlightedRanges =>
+                            highlightedRanges.map(highlightedRange =>
+                                highlightedRange
```

Comment:
According to my LSIF hover this is never falsy?

![image](https://user-images.githubusercontent.com/10532611/86377061-21640a80-bc88-11ea-8c8c-02a9f09765d6.png)